### PR TITLE
Use restCall object for functions that makes requests

### PR DESCRIPTION
**Pull request by rusan**

### CODE REVIEW

File: lib/PayPal/Api/FuturePayment.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ class FuturePayment extends Payment
      * @param string|null  $clientMetadataId
      * @return $this
      */
-    public function create($apiContext = null, $clientMetadataId = null)
+    public function create($apiContext = null, $clientMetadataId = null, $call = null)
```
Comment (jaypatel512) [inline lib/PayPal/Api/FuturePayment.php]: Can we please name this `$restCall` too, just for consistency.

File: lib/PayPal/Api/FuturePayment.php [hunk 2]
```diff
@@ -32,7 +32,7 @@ public function create($apiContext = null, $clientMetadataId = null)
             );
         }
         $payLoad = $this->toJSON();
-        $call = new PayPalRestCall($apiContext);
+        $call = $call ? $call : new PayPalRestCall($apiContext);
```
Comment (jaypatel512) [inline lib/PayPal/Api/FuturePayment.php]: Curious, if we could just replace this with `executeCall` itself. Maybe we can do that as a separate PR.

**Comment by rusan** [comment]
@jaypatel512 - Done.

**Comment by jaypatel512** [comment]
You are awesome @rusan ! 

We will do a quick test on this PR, and merge it soon. Thank you again for your contribution. 🎆
